docs/DeveloperGuide.md

```diff
@@ -4,35 +4,186 @@
+### Flower Help Command
 
-## Non-Functional Requirements
+`help` Shows a list of valid commands
 
-{Give non-functional requirements}
+![Help Command Sequence Diagram](UML-diagrams/Jeffinson/Jeffinson-UML-Help.png);
 
-## Glossary
+Help command utilises the `parser` class to identify the keyword `help`
 
-* *glossary item* - Definition
+**Step 1: ** User will write `help` and `HelpCommand` class will be instantiated and `execute()` method will run.
 
-## Instructions for manual testing
+**Step 2:** `HelpCommand` class will call `printHelpMessage()` method of `Ui` class
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+**Step 3:** The list of valid commands will be printed by the `Ui` class
+
+### Add Flower Command
+
+`add <flower> /q <quantity> /to <targetBouquet>` command adds specified number of flower to a bouquet
+
+![Add Flower Command Diagram](/docs/UML-diagrams/Ijaaz/Ijaaz-UML.png)
+
+Step 1: Firstly the input is put into the `Parser.parse()` method to identify that it is infact an add flower command.
+
+Step 2: Then an object of type `AddFlowerCommand` is returned which contains the name, quantity of the flower to be added, as well as the target bouquet
+
+Step 3: The `execute()` method is called to execute the add flower command 
+
+Step 4: The target bouquet, which is under model, is updated accordingly
+
+Step 5: A confirmation message is then sent back to the user
+
+### [Proposed] Storage
+
+#### Proposed Implementation:
+
+The proposed storage mechanism will utilize a class `storage` who is in charge of getting the file, `encoder` which will encode current model into a .txt file format, `decoder` which will decode .txt file into a usable model.
+
+![Storage Class Diagram](/docs/UML-diagrams/Ian/storage.png)
+
+#### Design Considerations:
+- Alternative 1 (current choice): use .txt files as storage
+    - Pros:
+        - Easier to implement
+        - .txt file is very readable even outside of programme
+    - Cons:
+        - .txt file can be easily changed outside of programme, making storage of model prone to parsing failure
+- Alternative 2: use .csv files as storage
+    - Pros:
+        - A more standardize format that is easily accepted by other 3rd party software
+    - Cons:
+        - More difficulty to set up the decoder
+        - .csv file can be easily changed outside of programme, making storage of model prone to parsing failure
+
+
+# Appendix: Requirements
+
+## Product scope
+
+### Target user profile:
+- has a need of purchasing a bouquet
+- prefers desktop apps over other types
+- can type reasonably fast
+- prefers typing to mouse interactions
+- is comfortable using CLI apps
+
+### Value proposition:
+- organise flowers faster than using a GUI driven app
+- curate flowers automatically based on context given through typed text
+
+## User Stories
+Priorities: High (must have) `* * *`, Medium (nice to have) `* *`, Low (unlikely to have) `*`
```

Comment:
So are all the user stories here already implemented? Not very clear about that?